File: scripts/test_solutions/128_longest_consecutive.py
def longestConsecutive(nums):
    num_set = set(nums)
    max_len = 0
    for num in num_set:
        if num - 1 not in num_set:
            curr = num
            length = 1
            while curr + 1 in num_set:
                curr += 1
                length += 1
            max_len = max(max_len, length)
    return max_len

File: scripts/test_solutions/128_longest_consecutive.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int maxLen = 0;
        for (int num : numSet) {
            if (numSet.find(num - 1) == numSet.end()) {
                int curr = num;
                int length = 1;
                while (numSet.find(curr + 1) != numSet.end()) {
                    curr++;
                    length++;
                }
                maxLen = max(maxLen, length);
            }
        }
        return maxLen;
    }
};
